odl/solvers/nonsmooth/proximal_operators.py

```diff
@@ -248,14 +264,20 @@ def proximal_arg_scaling(prox_factory, scaling):
     2011.
     """
 
-    if scaling.imag != 0:
+    # To begin, we could check for two things:
+    # * Currently, we do not support complex scaling. We could therefore catch
+    #   nonempty imaginary parts.
+    # * If some components of scaling are zero, then the following routine will
+    #   crash with a division-by-zero error. The correct solution would be to
+    #   just keep these components and do the following computations only for
+    #   the others.
+    # Since these checks are computationally expensive, we do not execute them
+    # unconditionally, but only if the scaling factor is a scalar:
+    if np.isscalar(scaling) and scaling.imag != 0:
         raise ValueError("complex scaling not supported.")
-    else:
-        scaling = float(scaling.real)
-    if scaling == 0:
+    if np.isscalar(scaling) and scaling == 0:
         return proximal_const_func(prox_factory(1.0).domain)
```

Comment:
You need an `else` with conversion to an array here, otherwise [this line](https://github.com/odlgroup/odl/pull/1166/files#diff-96685646d224192b17008ecba5b0eab1R295) will crash.